reflect/src/main/java/dagger/reflect/Scope.java

```diff
@@ -58,6 +59,16 @@ private Scope(ConcurrentHashMap<Key, Binding> bindings, JustInTimeLookup.Factory
         Key realKey = Key.of(key.qualifier(), parameterizedKeyType.getActualTypeArguments()[0]);
         return new LinkedLazyBinding<>(this, realKey);
       }
+      if (rawKeyType == Map.class) {
+        if (parameterizedKeyType.getActualTypeArguments()[1] instanceof ParameterizedType) {
```

Comment:
Is there a test for a simple provided `Map`?
```java
@Provides static Map<String, Number> providesMap() {
    Map<String, Number> result = new HashMap<>();
    result.put("one", 1);
    result.put("two", 2.0);
    return result;
}
```
just in case the above condition breaks that.

Reply:
excellent idea! i have made a separate PR for this: https://github.com/JakeWharton/dagger-reflect/pull/112